src/plugins/data/public/query/query_string/query_string_manager.ts

```diff
@@ -63,6 +63,16 @@ export class QueryStringManager {
     return this.storage.get('userQueryString') || '';
   }
 
+  private getInitialDatasetQueryString(query: Query, dataset: Dataset) {
```

Comment:
The reason to have the additional `dataset` parameter is to avoid unnecessary checks for `query.dataset` since all the caller methods call this only with valid dataset

Reply:
It it possible for dataset to not match query.dataset? If this is not checked for by calling code, can we throw when these do not match as an invariant?

Reply:
Just in case I'm not around, this is the only real blocking comment to approval from me - just worried about a potential mismatch between dataset and query's dataset - if there's something that prevents that situation, or we enforce we don't get into that situation, I'm good.

Reply:
updated to just use the dataset from the query